src/components/Tree/TreeItem/TreeItemOverlay.tsx

```diff
@@ -1,27 +1,22 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import React, { Children } from 'react';
+import React, { Children, ReactNode } from 'react';
 
 import { merge } from '@utilities/merge';
 import { IconGrabHandle12 } from '@foundation/Icon';
 
 import { INDENTATION_WIDTH } from '../Tree';
-import type { TreeItemProps } from '../types';
-import { useTreeContext } from '../TreeContext';
 
-export type Overlay = Pick<TreeItemProps, 'id' | 'label' | 'contentComponent' | 'children' | 'level'>;
-
-export const TreeItemOverlay = () => {
-    const { treeState } = useTreeContext();
-
-    if (!treeState.overlay) {
-        return null;
-    }
-
-    const { id, children, label, level = 0, contentComponent } = treeState.overlay;
-
-    const isSelected = treeState.selectedIds.has(id);
+export type Overlay = {
+    id: string;
+    label?: string;
+    level?: number;
+    children: ReactNode;
+    contentComponent?: ReactNode;
+    isSelected?: boolean;
+};
 
+export const TreeItemOverlay = ({ id, label, level = 0, children, contentComponent, isSelected }: Overlay) => {
```

Comment:
regarding `level`: `aria-level >= 1` like in _treeItem_ or remove it if it's not needed 

```
aria-level={level + 1}
```